### DISCUSSION TITLE
mpi meep

### DISCUSSION THREAD

**Discussion by efrenaserra** [NanoComp/meep#2545] [General]
Folks - I'm having issues after following the installation instructions at: https://meep.readthedocs.io/en/latest/Installation/, both meep and parallel meep. Neither execution of straight-waveguide.py ever finishes. Also, I get the following error when I do as follows:

1] conda activate pmp
2] mpirun -n 4 python3 $HOME/pymeep/python/examples/straight-waveguide.py

Here's the error:
mpiexec.hydra -n 4 python3 $HOME/pymeep/python/examples/straight-waveguide.py
(base) [serra@archer ~]$ conda activate pmp
(pmp) [serra@archer ~]$ bash -x pmp.job
+ mpiexec.hydra -n 4 python3 /users/serra/pymeep/python/examples/straight-waveguide.py
Using MPI version 4.0, 4 processes
-----------
Initializing structure...
Splitting into 4 chunks by voxels
time for choose_chunkdivision = 0.000222206 s
Working in 2D dimensions.
Computational cell is 16 x 8 x 0 with resolution 10
     block, center = (0,0,0)
          size (1e+20,1,1e+20)
          axes (1,0,0), (0,1,0), (0,0,1)
          dielectric constant epsilon diagonal = (12,12,12)
time for set_epsilon = 0.00536799 s
-----------
libGL error: unable to load driver: swrast_dri.so
libGL error: failed to load driver: swrast
libGL error: unable to load driver: swrast_dri.so
libGL error: failed to load driver: swrast
libGL error: unable to load driver: swrast_dri.so
libGL error: failed to load driver: swrast
libGL error: unable to load driver: swrast_dri.so
libGL error: failed to load driver: swrast

**Comment by oskooi** [comment]
The error is related to [`libGL`](https://dri.freedesktop.org/wiki/libGL/) which is *not* a [Meep dependency](https://meep.readthedocs.io/en/latest/Build_From_Source/#required-dependencies). The problem could be due to a package conflict in one of the [Meep dependencies](https://meep.readthedocs.io/en/latest/Build_From_Source/#required-dependencies) installed in your system directories. For example, your system may already contain an MPI library which conflicts with the [mpich](https://github.com/conda-forge/pymeep-feedstock/blob/main/recipe/conda_build_config.yaml#L3) used by the Meep Conda package.

You can try installing the Conda package or [building Meep from source](https://meep.readthedocs.io/en/latest/Build_From_Source) on a clean machine (e.g., a virtual machine in the public cloud running Ubuntu).
